synapse/lib/httpapi.py

```diff
@@ -149,45 +151,132 @@ async def reqAuthUser(self):
+        udef = await authcell.getUserDef(iden, packroles=False)
+        if not udef.get('admin'):
+            return False
+
+        return True
+
     async def reqAuthAdmin(self):
+        '''
+        Require the current authenticated user to be an admin.
 
-        user = await self.user()
-        if user is None:
+        Notes:
+            If this returns False, an error message has already been sent
+            and no additional processing for the request should be done.
+
+        Returns:
+            bool: True if the user is an admin, false otherwise.
+        '''
+        iden = await self.useriden()
+        if iden is None:
             self.sendAuthRequired()
             return False
 
-        if not user.isAdmin():
-            self.sendRestErr('AuthDeny', f'User {user.iden} ({user.name}) is not an admin.')
+        authcell = self.getAuthCell()
+        udef = await authcell.getUserDef(iden, packroles=False)
+        if not udef.get('admin'):
+            self.sendRestErr('AuthDeny', f'User {self.web_useriden} ({self.web_username}) is not an admin.')
             return False
 
         return True
 
     async def sess(self, gen=True):
+        '''
+        Get the heavy Session object for the request.
+
+        Args:
+            gen (bool): If set to True, generate a new session if there is no sess cookie.
+
+        Notes:
+            This stores the identifier in the ``sess`` cookie for with a 14 day expiration, stored
+            in the Cell.
+            Valid requests with that ``sess`` cookie will resolve to the same Session object.
+
+        Returns:
+            Sess: A heavy session object. If the sess cookie is invalid or gen is false, this returns None.
+        '''
 
         if self._web_sess is None:
 
             iden = self.get_secure_cookie('sess', max_age_days=14)
 
-            if iden is None and not gen:
-                return None
-
             if iden is None:
-                iden = s_common.guid().encode()
-                opts = {'expires_days': 14, 'secure': True, 'httponly': True}
-                self.set_secure_cookie('sess', iden, **opts)
+                if gen:
+                    iden = s_common.guid().encode()
+                    opts = {'expires_days': 14, 'secure': True, 'httponly': True}
+                    self.set_secure_cookie('sess', iden, **opts)
+                else:
+                    return None
 
             self._web_sess = await self.cell.genHttpSess(iden)
 
         return self._web_sess
 
-    async def user(self):
+    async def user(self):  # pragma: no cover
+        '''
+        Get the heavy User object from the Cell for the authenticated user.
+
+        Deprecated, use the useriden() function instead.
+        '''
+        s_common.deprecated('HandlerBase.user() - use useriden()', eolv='v2.130.0')  # set curv?
+        iden = await self.useriden()
+        if iden is None:
+            return None
+        user = self.cell.auth.user(iden)
```

Comment:
Should this be using getAuthCell?

Reply:
We can't use that in this handler; since the cell represented by getAuthCell may not be an actual Cell implementation ( such as a telepath proxy or client object ), so we cannot deference things off of it.